kazuki-ksp/src/main/kotlin/com/anaplan/engineering/kazuki/ksp/type/MappingType.kt

```diff
@@ -187,6 +188,56 @@ private fun TypeSpec.Builder.addMappingType(
             interfaceClassDcl, typeGenerationContext, additionalInvariantParts
         )
 
+        if (!interfaceClassDcl.hasSuperType(PrettyPrintable::class.qualifiedName!!)) {
```

Comment:
⁉️ I'm sure I'm missing something trivial here... it looks like this is saying "if we _don't_ have `PrettyPrintable` as supertype, then override `pretty`", but surely it should be the other way around (override only if we have `PrettyPrintable` as supertype), shouldn't it?

Reply:
Unless what this is actually doing is providing a default implementation of `pretty` without overriding a previously defined one if present, but then shouldn't it also add `PrettyPrintable` as supertype to have something to override?

Reply:
The impl class will implement the interface and also (indirectly) `_KazukiObject` 

`_KazukiObject`  is a `PrettyPrintable`

If we add `pretty' to the impl it would overrides the version provided by the user in the interface.. so we only want to add if the interface is not  `PrettyPrintable`. (But we want all `_KazukiObject`  to be `PrettyPrintable`)